pkg/orm/client.go

```diff
@@ -0,0 +1,610 @@
+package orm
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	dmysql "github.com/go-sql-driver/mysql"
+	libModel "github.com/hanfei1991/microcosm/lib/model"
+	cerrors "github.com/hanfei1991/microcosm/pkg/errors"
+	"github.com/hanfei1991/microcosm/pkg/externalresource/resourcemeta"
+	"github.com/hanfei1991/microcosm/pkg/meta/metaclient"
+	"github.com/hanfei1991/microcosm/pkg/orm/model"
+	"github.com/hanfei1991/microcosm/pkg/tenant"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+)
+
+// TODO: retry and idempotent??
+// TODO: context control??
+
+type TimeRange struct {
+	start time.Time
+	end   time.Time
+}
+
+type Client interface {
+	metaclient.Client
+
+	// Initialize will create all tables for backend operation
+	Initialize(ctx context.Context) error
+
+	// project
+	AddProject(ctx context.Context, project *model.ProjectInfo) error
+	DeleteProject(ctx context.Context, projectID string) error
+	QueryProjects(ctx context.Context) ([]*model.ProjectInfo, error)
+	GetProjectByID(ctx context.Context, projectID string) (*model.ProjectInfo, error)
+
+	// project operation
+	AddProjectOperation(ctx context.Context, op *model.ProjectOperation) error
+	QueryProjectOperations(ctx context.Context, projectID string) ([]*model.ProjectOperation, error)
```

Comment:
Do we need pagination and cursor here?

Reply:
Thanks. I will add a TODO here and do it later.